pydantic/functional_serializers.py

```diff
@@ -217,6 +259,30 @@ def model_serializer(
 ) -> Callable[[Any], Any]:
     """Decorator that enables custom model serialization.
 
+    This is useful when a model need to be serialized in a customized manner, allowing for flexibility beyond just specific fields.
+
+    An example would be to serialize temperature to the same temperature scale, such as degrees Celsius.
+
+    ```python
+    from typing import Literal
+
+    from pydantic import BaseModel, model_serializer
+
+    class TemperatureModel(BaseModel):
+        unit: Literal['C', 'F']
+        value: int
+
+        @model_serializer()
+        def serialize_model(self):
+            if self.unit == 'F':
+                return {'unit': 'C', 'value': int((self.value - 32) / 1.8)}
+            return {'unit': self.unit, 'value': self.value}
+
+    temperature = TemperatureModel(unit='F', value=212)
+    print(temperature.model_dump())
+    #> {'unit': 'C', 'value': 100}
+    ```
```

Comment:
Perfect example! Love this change.

Reply:
Glad to hear that.
The remaining is `WrapSerializer` which iam working on.

Reply:
Fantastic!